lib/logstash/codecs/cef.rb

```diff
@@ -261,10 +261,20 @@ def initialize(params={})
   def decode(data, &block)
     if @delimiter
       @buffer.extract(data).each do |line|
-        handle(line, &block)
+        dumping_handle(line, &block)
       end
     else
+      dumping_handle(data, &block)
+    end
+  end
+
+  def dumping_handle(data, &block)
+    begin
+      original_data = data.dup
       handle(data, &block)
+    rescue => e
+      logger.error("Problem processing data : ", :original_data => original_data)
+      raise
```

Comment:
the `begin` `rescue` and `logger` sounds simple enough to merge in `handle()`
Any reason not to enhance `handle()` and add log in [rescue](https://github.com/logstash-plugins/logstash-codec-cef/blob/master/lib/logstash/codecs/cef.rb#L337) ?

Reply:
Nice suggestion!